sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorBuilder.java

```diff
@@ -0,0 +1,187 @@
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.eventhubs.models.EventPosition;
+import reactor.core.publisher.Mono;
+
+/**
+ * This class provides a fluent builder API to help aid the configuration and instantiation of the {@link
+ * EventProcessor}. Calling {@link #buildEventProcessor()} constructs a new instance of {@link EventProcessor}.
+ *
+ * <p>
+ * To create an instance of {@link EventProcessor} that processes events with user-provided callback, configure the
+ * following fields:
+ *
+ * <ul>
+ * <li><b>Consumer group name.</b></li>
+ * <li><b>{@link EventHubAsyncClient}</b> - An asynchronous Event Hub client the {@link EventProcessor} will use for
+ * consuming events.</li>
+ * <li><b>{@link #processEvent(ProcessEventConsumer)}</b> - A callback for processing new events as they are received.</li>
+ * <li><b>{@link PartitionManager}</b> - An instance of PartitionManager. To get started, you can pass an instance of
+ * {@link InMemoryPartitionManager}. For production, choose an implementation that will store checkpoints and partition
+ * ownership details to a durable store.</li>
+ * </ul>
+ *
+ * <p><strong>Creating an {@link EventProcessor} using a {@link #processEvent(ProcessEventConsumer)} function</strong></p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.processevent}
+ *
+ * <p>
+ *  To create a more advanced {@link EventProcessor}, use the
+ *  {@link #partitionProcessorFactory(PartitionProcessorFactory)} instead of {@link #processEvent(ProcessEventConsumer)}.
+ *  The {@link PartitionProcessorFactory} provides the ability to control when checkpoints are updated, handle
+ *  errors that might occur during event processing and perform partition-specific initialization of
+ *  {@link PartitionProcessor}.
+ * </p>
+ *
+ * <p><strong>Creating an {@link EventProcessor} using {@link PartitionProcessorFactory}</strong></p>
+ * {@codesnippet com.azure.messaging.eventhubs.eventprocessorbuilder.partitionprocessorfactory}
+ *
+ * @see EventProcessor
+ * @see EventHubConsumer
+ */
+public class EventProcessorBuilder {
+
+    private final ClientLogger logger = new ClientLogger(EventProcessorBuilder.class);
+
+    private EventPosition initialEventPosition;
+    private PartitionProcessorFactory partitionProcessorFactory;
+    private String consumerGroup;
+    private PartitionManager partitionManager;
+    private EventHubAsyncClient eventHubAsyncClient;
+    private ProcessEventConsumer processEvent;
+
+    /**
+     * Sets the Event Hub client the {@link EventProcessor} will use to connect to the Event Hub for consuming events.
+     *
+     * @param eventHubAsyncClient The Event Hub asynchronous client for consuming events.
+     * @return The updated {@link EventProcessorBuilder} instance.
+     */
+    public EventProcessorBuilder eventHubClient(EventHubAsyncClient eventHubAsyncClient) {
```

Comment:
For the EventHubClient, we could pass in an overload that accepts the synchronous one? It wouldn't matter to us since we can access that underlying async client internally, but it'll make it so that users don't have to construct another if they want to synchronously send.

Reply:
As discussed, deferring it for later. Synchronous client may lead the user to believe that the event processor is going to use synchronous APIs even though we can get access to the async client underneath. We can discuss this option for P4.